### ISSUE TITLE
[BUG] `rancher-monitoring` in Harvester for New User Creation, not providing correct URL path for `invite`

### ISSUE THREAD

**Issue by irishgordo** [harvester/harvester#5298]
**Describe the bug**
If a user/admin/org/etc. is trying to leverage the underlying [Rancher Monitoring stack](https://ranchermanager.docs.rancher.com/how-to-guides/advanced-user-guides/monitoring-alerting-guides/customize-grafana-dashboard) w/ Grafana & Prometheus there is an issue where the new user invite isn't providing the correct URL, if they are to have additional users creating metrics/dashboards on Grafana for their Harvester cluster.

**To Reproduce**
1. Be on Harvester Dashboard
1. Click call-to-action/button to take us to Grafana landing page (rancher-monitoring)
1. Login [step 5, w/ admin/prom-operator](https://ranchermanager.docs.rancher.com/how-to-guides/advanced-user-guides/monitoring-alerting-guides/customize-grafana-dashboard)
1. Click the gear-icon/button -> configuration -> then select 'users' tab (x-ref screengrab)
1. build a new user, no email
1. click the `copy invite` 
1. notice that when pasted in the browser it is directing somewhere in-accessible by the user example: `http://localhost:3000/invite/3qq7sUO9n7MF3XfRH8YfDWR94pYhli`

**Workaround**
1. the user needs to manually shift the uri/url pattern to be more appropriate of something like instead: `https://192.168.12.64/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-grafana:80/proxy/invite/3qq7sUO9n7MF3XfRH8YfDWR94pYhli`

**Expected behavior**
Harvester VIP + rancher-monitoring-grafana info to be prepended to "invite" that is generated:
```
https://192.168.12.64/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-grafana:80/proxy/
```
+ the invite info.

**Environment**
 - Harvester ISO version: v1.3-head /  Version: v1.3-4372cbc4-head 
 - Underlying Infrastructure: 3 node hp dl160 gen 9s

**Additional context**
![Screenshot from 2024-03-05 11-22-03](https://github.com/harvester/harvester/assets/5370752/5800edf9-0dbf-49d1-a3fe-3292a6b6be5b)
![Screenshot from 2024-03-05 11-12-00](https://github.com/harvester/harvester/assets/5370752/1e495fea-9f7e-41da-94c6-aeffe2137a32)
![Screenshot from 2024-03-05 11-13-05](https://github.com/harvester/harvester/assets/5370752/4b832bb2-03b9-4cd9-8834-4429c57c35b0)
![Screenshot from 2024-03-05 11-18-08](https://github.com/harvester/harvester/assets/5370752/36ba4179-d2b1-455c-b8d6-78c2b0aee029)

**Comment by irishgordo** [comment]
@w13915984028  - do you happen to know if we're missing Prometheus configuration to fix this?

I'm still seeing this w/ v1.4.2

Invite links still are: http://localhost:3000/invite/HIs9kI0VuUw45YdckVl3UnBezpPJV4

Versus, having the IPv4 address in the URL.

Also w/ Harvester imported into Rancher, it behaves the same way too.

**Comment by Vicente-Cheng** [comment]
Since this issue was very old, @w13915984028 do you think this issue is still here?
Thanks!

**Comment by irishgordo** [comment]
@Vicente-Cheng @w13915984028  this is still even reproducible in v1.6.1:

```
http://localhost:3000/invite/smVzjAUMSx0WKfl0giG1w0zEQ0d6ID
```

^ is the Invite URL that's provided.... 


<img width="2032" height="2021" alt="Image" src="https://github.com/user-attachments/assets/3f8d6040-b9ac-4eec-900f-588ac94a6c37" />
<img width="2049" height="2074" alt="Image" src="https://github.com/user-attachments/assets/347747f9-b4d0-4a86-bf28-2bd3931b2311" />
<img width="2037" height="1995" alt="Image" src="https://github.com/user-attachments/assets/356b878d-e03f-48f2-8f50-5305c5d70a42" />

**Comment by irishgordo** [comment]
Though newer Grafana you need to:
1. Invite a user, no email, build a user
2. Go to organizational users
3. Select Copy Invite

Then that's where you'll see `localhost`

**Comment by irishgordo** [comment]
The workaround URL would look like:

```
https://10.115.16.193/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-grafana:80/proxy/invite/smVzjAUMSx0WKfl0giG1w0zEQ0d6ID
```

Not:

```
http://localhost:3000/invite/smVzjAUMSx0WKfl0giG1w0zEQ0d6ID
```

And then the user is able to "see" the invite and utilize it.

<img width="1503" height="2046" alt="Image" src="https://github.com/user-attachments/assets/55e206d8-a3bc-4a90-a358-ae9f2ea5b8fa" />

**Comment by w13915984028** [comment]
@irishgordo @Vicente-Cheng 

This can be reproduced on Harvester v1.7.1.

The root cause is that the generated link is not sent directly, hence the internal proxy can't append the prefix to generate a full external visitable URL, the workaround mentioned by @irishgordo works. And this is fully done by grafana. What we can do is to write it down.

Let me add this to the document. thanks.

**Comment by ihcsim** [comment]
Unless there is a clear need for it, I think we should just disable Grafana's [basic password login](https://github.com/rancher/charts/blob/274b7b65bda65df3c904e80d1c52f567aac364bb/charts/rancher-monitoring/108.0.2%2Bup77.9.1-rancher.11/values.yaml#L1940-L1941) because Rancher already handles [access to Grafana via RBAC](https://ranchermanager.docs.rancher.com/integrations-in-rancher/monitoring-and-alerting/rbac-for-monitoring).

**Comment by irishgordo** [comment]
Yeah, I have no idea the scale of users that may be building other users aside of `admin` / `prom-operator` and using those users for dashboard building / custom stuff 🤔

**Comment by ihcsim** [comment]
The grafana users setup also don't persist; they'll be wiped the moment the monitoring add-on is disabled. I recommend closing this issue as "won't fix". Then create a separate issue to discuss whether to disable Grafana's basic password login.

**Comment by irishgordo** [comment]
I'm in favor of that then too 😄  ^

**Comment by irishgordo** [comment]
I think the "doc" might not hurt things, but it might be worthwhile mentioning the fact that the stuff won't persist if it's disabled

**Comment by w13915984028** [comment]
When `rancher-monitoring` addon is disabled, the related PVC used by `prometheus,grafana,alertmanager` are not removed, they are kept and will be picked up by default if addon is enabled again.

I did a test, both the added user and pending invite are kept after addon is re-enabled.

<img width="2870" height="932" alt="Image" src="https://github.com/user-attachments/assets/a641c2d2-e83b-42a0-9101-6ae7883c29cd" />

pending invite
<img width="3590" height="718" alt="Image" src="https://github.com/user-attachments/assets/235fb983-b9df-47fa-9123-5cd08d38159e" />

This bug was not reported in the past by customer, it might be related to: for most production deployments, a central monitoring/logging system is set, and Harvester's monitoring/logging are connected to external. The embedded grafana is not extensively used.

Anyway, we could add this to document, and remind user better to disable password.

@irishgordo @ihcsim WDYT? thanks.

**Comment by irishgordo** [comment]
Ohhh so the PVC sticks around and data will persist if new users were created, graphs built, etc.

**Comment by w13915984028** [comment]
@irishgordo It is in general correct. The attached PVC is saved for reuse.

**Comment by ihcsim** [comment]
I maintain that we shouldn't support nor document this, and rather, (create a separate issue to) disable it completely. AFAIK, Harvester's only user management story is via Rancher integration. Users shouldn't use an unsupported method to set up user accounts in an add-on within the system.

**Comment by w13915984028** [comment]
I have no strong option to doc or close ticket. 

Close the doc PR, move this ticket to `won't fix` is fine, just keep things as is. If someone tris to play with embedded Grafana, this ticket could still be useful.

@irishgordo @Vicente-Cheng WDYT? thanks.

**Comment by Vicente-Cheng** [comment]
cc @ibrokethecloud, @innobead 

I agree that users shouldn't use an unsupported way to set up a user account in an add-on within the system, as @ihcsim mentioned.
But we don't have a way to restrict users from doing so.

I think we should at least show a main notification telling the user not to do that.
WDYT?

**Comment by w13915984028** [comment]
Mike's tests are upon the embedded grafana standalone UI, we don't have a normal way to show such notification. ATM, we enusres the main functions of Grafana work well, but don't touch all those corner cases.

Keeps things as is would be fine for now, if in future some customers (they normally deploy a centralized monitoring server like SUSE Observability) raise issues/requirements, we further work on them.

 @Vicente-Cheng @irishgordo @ihcsim WDYT? thanks.

**Comment by ihcsim** [comment]
closing as won't fix. see https://github.com/harvester/harvester/issues/10152.

**Comment by irishgordo** [comment]
yeah, this makes sense to close - thanks for all the cycles on this @ihcsim @w13915984028  👍 😄  - 10152 will be good
